src/topology/connected.lean

```diff
@@ -215,28 +215,32 @@ theorem is_connected.Union_of_refl_trans_gen {ι : Type*} [nonempty ι] {s : ι
 ⟨nonempty_Union.2 $ nonempty.elim ‹_› $ λ i : ι, ⟨i, (H _).nonempty⟩,
   is_preconnected.Union_of_refl_trans_gen (λ i, (H i).is_preconnected) K⟩
 
-/-- The Union of connected sets indexed by `ℕ` such that any two neighboring sets meet
-is preconnected. -/
-theorem is_preconnected.Union_nat_of_chain {s : ℕ → set α}
-  (H : ∀ n : ℕ, is_preconnected (s n))
-  (K : ∀ n : ℕ, (s n ∩ s n.succ).nonempty) :
+section succ_order
+open succ_order
+
+variables [linear_order β] [succ_order β] [is_succ_archimedean β]
+
+/-- The Union of connected sets indexed by a type with an archimedian successor (like `ℕ` or `ℤ`)
+  such that any two neighboring sets meet is preconnected. -/
+theorem is_preconnected.Union_of_chain {s : β → set α}
```

Comment:
I would get rid of dot notation here and below because you can't actually use it.

Reply:
I agree you cannot actually use it (though I kinda hope we can change the way projection notation works in Lean 4 so that `H.Union_of_chain K` works in this case).
However, I'm not sure if that means we should change the `.` to a `_`: it might be more predictable to guess the name if we consistently keep using `.` even when projection notation doesn't apply.

Reply:
At least, there's precedent for not using it when the condition is in a forall. For example, `star_convex_Union`.

Reply:
I believe there is a bunch of precedent for both options.